python/lsst/resources/http.py

```diff
@@ -368,8 +457,9 @@ def _make_session(self, rpath: ResourcePath) -> requests.Session:
             # How many times to retry on read errors.
             read=3,
             # Backoff factor to apply between attempts after the second try
-            # (seconds)
-            backoff_factor=30 * (1 + random.random()),
+            # (seconds). Compute a random jitter to prevent all the clients
+            # to overwhelm the server by sending requests at the same time.
+            backoff_factor=self._backoff_min + (self._backoff_max - self._backoff_min) * random.random(),
```

Comment:
The urllib3 implementation of retries (used by the requests package) does not add jitter to the backoff factor passed as argument. I thought it would be more flexible to allow configuring via environment variables an interval from when the backoff is computed including jitter. In the current implementation jitter is added but the backoff factor is kind of fixed to be in the interval [30.0, 60.0) seconds.